lib/yardstick/decorator.rb

```diff
@@ -0,0 +1,60 @@
+module Yardstick
+  # A string decorator for applying unix console codes
+  class Decorator
+    FORMAT = "\e[%{mode};%{color}%{string}\e[0m".freeze
+
+    COLOR_CODES = {
+      red:    31,
+      yellow: 33
+    }.freeze
+
+    MODE_CODES = {
+      bold:      1,
+      underline: 4
+    }.freeze
+
+    private_constant(:FORMAT, :COLOR_CODES, :MODE_CODES)
+
+    # Initializes new string decorator instance
+    #
+    # @param color [Symbol] name of color to use for formatting
+    # @param mode [Symbol] mode for formatting text
+    #
+    # @return [undefined]
+    #
+    # @api private
+    def initialize(color, mode)
+      @color = COLOR_CODES.fetch(color)
+      @mode  = MODE_CODES.fetch(mode)
+    end
```

Comment:
> you do have less chance to typo the ivars.

I'm adding in concord as a dependency. Would it also be worth it to update older code where ivars are used instead of `attr_reader`?  `RuleConfig` would be an example here